rego/Drive.rego

```diff
@@ -32,37 +54,61 @@ tests contains {
 }
 if {
     DefaultSafe := false
-    Events := utils.FilterEvents(LogEvents, "SHARING_OUTSIDE_DOMAIN", utils.TopLevelOU)
+    Events := utils.FilterEventsOU(LogEvents, "SHARING_OUTSIDE_DOMAIN", utils.TopLevelOU)
     count(Events) == 0
 }
 
 tests contains {
     "PolicyId": "GWS.DRIVEDOCS.1.1v0.1",
     "Criticality": "Should",
-    "ReportDetails": utils.ReportDetailsOUs(NonCompliantOUs1_1),
-    "ActualValue": {"NonCompliantOUs": NonCompliantOUs1_1},
+    "ReportDetails": utils.ReportDetails(NonCompliantOUs1_1, NonCompliantGroups1_1),
+    "ActualValue": {"NonCompliantOUs": NonCompliantOUs1_1, "NonCompliantGroups": NonCompliantGroups1_1},
     "RequirementMet": Status,
     "NoSuchEvent": false
 }
 if {
     Events := utils.FilterEvents(LogEvents, "SHARING_OUTSIDE_DOMAIN", utils.TopLevelOU)
     count(Events) > 0
-    Status := count(NonCompliantOUs1_1) == 0
+    Conditions := {count(NonCompliantOUs1_1) == 0, count(NonCompliantGroups1_1) == 0 }
+    Status := (false in Conditions) == false
 }
 #--
 
 # Can be combined with 1.1, since this is a single setting with the same value that will pass for both conditions
 #
 # Baseline GWS.DRIVEDOCS.1.2v0.1
 #--
-NonCompliantOUs1_2 contains OU if {
+
+
+GetFriendlyValue1_2(Value) := "Users cannot recieve files outside the domain" if {
+    contains("SHARING_NOT_ALLOWED INHERIT_FROM_PARENT", Value) == true
+}
+else := "Users can recieve files outside the domain"
+
+NonCompliantOUs1_2 contains {
+    "Name": OU,
+    "Value": concat("", [GetFriendlyValue1_2(LastEvent.NewValue)])
+    }
+    if {
     some OU in utils.OUsWithEvents
-    Events := utils.FilterEvents(LogEvents, "SHARING_OUTSIDE_DOMAIN", OU)
+    Events := utils.FilterEventsOU(LogEvents, "SHARING_OUTSIDE_DOMAIN", OU)
     count(Events) > 0
     LastEvent := utils.GetLastEvent(Events)
     contains("SHARING_NOT_ALLOWED INHERIT_FROM_PARENT", LastEvent.NewValue) == false
 }
 
+NonCompliantGroups1_2 contains {
+    "Name": Group,
+    "Value": concat("", [GetFriendlyValue1_2(LastEvent.NewValue)])
+    }
+    if {
+    some Group in utils.GroupsWithEvents
+    Events := utils.FilterEventsGroup(LogEvents, "SHARING_OUTSIDE_DOMAIN", Group)
+    count(Events) > 0
+    LastEvent := utils.GetLastEvent(Events)
+    contains("SHARING_NOT_ALLOWED INHERIT_FROM_PARENT", LastEvent.NewValue) == false
+    }
```

Comment:
This logic is incorrect. Enumerating the settings, you can see that SHARING_OUTSIDE_DOMAIN has the following possible values (plus INHERIT_FROM_PARENT):
![image](https://github.com/cisagov/ScubaGoggles/assets/106177711/7d24a274-d01f-4e08-9e8b-9ce2ad768585)

Of those, the following all meet the requirements for 1.2:
- SHARING_NOT_ALLOWED
- TRUSTED_DOMAINS_ALLOWED
- TRUSTED_DOMAINS_ALLOWED_WITH_WARNING
- INHERIT_FROM_PARENT

But this rule currently only accepts SHARING_NOT_ALLOWED and INHERIT_FROM_PARENT. We need to add the missing two acceptable values.

Which also means that the comment "Can be combined with 1.1, since this is a single setting with the same value that will pass for both conditions" is inaccurate as this rule needs different logic.

Final nitpick: I don't like the `contains("big string with space-separated list of values"...` syntax. It would be more readable and less error prone to make that a list instead of a space-separated string (as done in GWS.DRIVEDOCS.1.3v0.1).